README.md

```diff
@@ -1,119 +1,180 @@
+Both D1 and D2 modes use same configuration registers, only difference in configuration is setting aad_select to T5838_AAD_SELECT_D1 or T5838_AAD_SELECT_D2.
 
-Whenever you want to update the version of `sdk-nrf` (also know simply as `NCS`)
-you need to keep one general thing in mind: you need to manually keep revisions
-of `sdk-nrf` and `sdk-zephyr` projects, as well as their imports in sync.
+To use AAD D mode we need to set aad_select to T5838_AAD_SELECT_D1 or T5838_AAD_SELECT_D2 and set aad_d_floor, aad_d_rel_pulse_min, aad_d_abs_pulse_min, aad_d_abs_thr and aad_d_rel_thr to desired values.
 
-1. Open `west.yml` file in `sdk-nrf` repository
-   ([link](https://github.com/nrfconnect/sdk-nrf)). Make sure that you select
-   correct tag from selection from top-left dropdown menu.
-2. Check what repos are under `sdk-zepyhr` project's `name-allowlist`, those
-   should match the repos in `west.yaml` of your project (and this template),
-   under `sdk-zepyhr` project's `name-allowlist`. Most of the time they should
-   be commented out, but depends.
-3. Check what other repositories appear under `sdk-zepyhr` project as standalone
-   projects, they start appearing around line 100, after `NCS repositories`
-   comment. This projects should match the repos in `west.yaml` of your project
-   (and this template), under `sdk-nrf` project's `name-allowlist`. Most of the
-   time they should be commented out, but again, this depends.
+AAD D modes are not yet tested with this driver, so please refer to t5838.h file to see what values can be set for each of the registers. Please also refer to [t5838 datasheet](https://invensense.tdk.com/wp-content/uploads/2023/06/DS-000383-T5838-Datasheet-v1.1.pdf) for more information on AAD D modes.
 
-When a new `sdk-nrf` version is released, some new repos as `NCS repositories`
-might appear or be moved into `sdk-zephyr`.
+### API functions
+
+Configuration function is defined as follows:
+
+```c
+int t5838_configure_AAD(const struct device *dev, struct T5838_thconf *thconf);
+```
+
+we pass pointer to our configuration struct as an argument. Function will write configuration to microphone registers and put microphone in desired AAD mode. This function must be called after calling `dmic_configure` pdm API function.
+
+Function for setting interrupt handler is defined as follows:
+
+```c
+void t5838_wake_set_handler(const struct device *dev, T5838_wake_handler_t handler);
+```
+
+This function sets handler to be called when wake interrupt is triggered. Note that interrupt will be called once and then it will be disabled. If you want to use wake interrupt again, you must call `t5838_wake_clear();` function.
+
+Function for clearing interrupt is defined as follows:
+
+```c
+int t5838_wake_clear(const struct device *dev);
+```
+
+This function clears wake interrupt and reenables it. It is used to reenable wake interrupt after it has been triggered.
+
+Function for getting wake pin state is defined as follows:
```

Comment:
Everything in **AAD API** section is redundant. You already describe everything in the header file. It is much better to have a single point of truth of documentation than two.